packages/lidar_map/include/lidar_map/builder.h

```diff
@@ -24,15 +23,22 @@ class Builder {
   public:
     Builder(const BuilderParams& params);
 
-    std::pair<geom::Poses, Clouds> filterByPosesProximity(
-        const geom::Poses& poses, const Clouds& clouds) const;
+    std::pair<geom::Poses, Clouds> sliceDataByPosesProximity(
+        const geom::Poses& poses, const Clouds& clouds, double poses_min_dist) const;
```

Comment:
Fixed bug in function `filterByPosesProximity`, now it's called `sliceDataByPosesProximity`.

It used RTree to filter points out (it was @dasimagin idea), but it has primarily two cons: firstly, for loop closure it might be better to have two very close to each other points, secondly if you take a ride with car on the same route, thinking that it will increase quality of resulting map, the second lap will be fully removed by this filter, because we will remember each point on the first lap, and will not include points from the second lap counting them as very close points. That's why I removed RTree and made back old filtration logic that looked at only two close by time points.

Thanks to @hexwhyzet for this information!